plotly/tests/test_optional/test_figure_factory.py

```diff
@@ -529,3 +530,451 @@ def test_dendrogram_colorscale(self):
+                              1.0,
+                              0.0,
+                              None,
+                              0.0,
+                              1.0,
+                              0.0,
+                              0.0,
+                              None,
+                              1.0,
+                              0.0,
+                              1.0,
+                              1.0,
+                              None],
+                             'z': [-0.0,
+                              -0.0,
+                              0.0,
+                              -0.0,
+                              None,
+                              -0.0,
+                              -0.0,
+                              1.0,
+                              -0.0,
+                              None,
+                              -0.0,
+                              0.0,
+                              0.0,
+                              -0.0,
+                              None,
+                              0.0,
+                              -0.0,
+                              -1.0,
+                              0.0,
+                              None,
+                              0.0,
+                              -0.0,
+                              0.0,
+                              0.0,
+                              None,
+                              -0.0,
+                              0.0,
+                              -1.0,
+                              -0.0,
+                              None,
+                              0.0,
+                              0.0,
+                              0.0,
+                              0.0,
+                              None,
+                              0.0,
+                              0.0,
+                              1.0,
+                              0.0,
+                              None]}],
+                           'layout': {'height': 800,
+                            'scene': {'aspectratio': {'x': 1, 'y': 1, 'z': 1},
+                             'xaxis': {'backgroundcolor': 'rgb(230, 230, 230)',
+                              'gridcolor': 'rgb(255, 255, 255)',
+                              'showbackground': True,
+                              'zerolinecolor': 'rgb(255, 255, 255)'},
+                             'yaxis': {'backgroundcolor': 'rgb(230, 230, 230)',
+                              'gridcolor': 'rgb(255, 255, 255)',
+                              'showbackground': True,
+                              'zerolinecolor': 'rgb(255, 255, 255)'},
+                             'zaxis': {'backgroundcolor': 'rgb(230, 230, 230)',
+                              'gridcolor': 'rgb(255, 255, 255)',
+                              'showbackground': True,
+                              'zerolinecolor': 'rgb(255, 255, 255)'}},
+                            'title': 'Trisurf Plot',
+                            'width': 800}}
```

Comment:
😿 Can you reformat this? You could even save on indentation by defining it at the top of the file or something. For example, this would be much better:

```
{
    'data': [
        {
            'facecolor': ['rgb(143.0, 123.0, 97.000000000000014)',
                          'rgb(255.0, 127.0, 14.000000000000007)',
                          'rgb(143.0, 123.0, 97.000000000000014)',
                          'rgb(31.0, 119.0, 180.0)',
                          'rgb(143.0, 123.0, 97.000000000000014)',
                          'rgb(31.0, 119.0, 180.0)',
                          'rgb(143.0, 123.0, 97.000000000000014)',
                          'rgb(255.0, 127.0, 14.000000000000007)'],
            'i': [3, 1, 1, 5, 7, 3, 5, 7],
            'j': [1, 3, 5, 1, 3, 7, 7, 5],
            'k': [4, 0, 4, 2, 4, 6, 4, 8],
            'name': '',
            'type': 'mesh3d',
            'x': np.array([-1.,  0.,  1., -1.,  0.,  1., -1.,  0.,  1.]),
            'y': np.array([-1., -1., -1.,  0.,  0.,  0.,  1.,  1.,  1.]),
            'z': np.array([ 1., -0., -1., -0.,  0.,  0., -1.,  0.,  1.])
        },
        {
        'line': {'color': 'rgb(50, 50, 50)', 'width': 1.5},
        'mode': 'lines',
        'type': 'scatter3d',
        'x': [-1.0, 0.0, 0.0, -1.0, None, 0.0, -1.0, -1.0, 0.0, None, 0.0,
              1.0, 0.0, 0.0, None, 1.0, 0.0, 1.0, 1.0, None, 0.0, -1.0, 0.0,
              0.0, None, -1.0, 0.0, -1.0, -1.0, None, 1.0, 0.0, 0.0, 1.0,
              None, 0.0, 1.0, 1.0, 0.0, None],
        'y': [0.0, -1.0, 0.0, 0.0, None, -1.0, 0.0, -1.0, -1.0, None, -1.0,
              0.0, 0.0, -1.0, None, 0.0, -1.0, -1.0, 0.0, None, 1.0, 0.0,
              0.0, 1.0, None, 0.0, 1.0, 1.0, 0.0, None, 0.0, 1.0, 0.0, 0.0,
              None, 1.0, 0.0, 1.0, 1.0, None],
        'z': [-0.0, -0.0, 0.0, -0.0, None, -0.0, -0.0, 1.0, -0.0, None,
              -0.0, 0.0, 0.0, -0.0, None, 0.0, -0.0, -1.0, 0.0, None, 0.0,
              -0.0, 0.0, 0.0, None, -0.0, 0.0, -1.0, -0.0, None, 0.0, 0.0,
              0.0, 0.0, None, 0.0, 0.0, 1.0, 0.0, None]
        }
    ],
    'layout': {
        'height': 800,
        'scene': {'aspectratio': {'x': 1, 'y': 1, 'z': 1},
        'xaxis': {'backgroundcolor': 'rgb(230, 230, 230)',
        'gridcolor': 'rgb(255, 255, 255)',
        'showbackground': True,
        'zerolinecolor': 'rgb(255, 255, 255)'},
        'yaxis': {'backgroundcolor': 'rgb(230, 230, 230)',
        'gridcolor': 'rgb(255, 255, 255)',
        'showbackground': True,
        'zerolinecolor': 'rgb(255, 255, 255)'},
        'zaxis': {'backgroundcolor': 'rgb(230, 230, 230)',
        'gridcolor': 'rgb(255, 255, 255)',
        'showbackground': True,
        'zerolinecolor': 'rgb(255, 255, 255)'}},
        'title': 'Trisurf Plot',
        'width': 800
    }
}
```